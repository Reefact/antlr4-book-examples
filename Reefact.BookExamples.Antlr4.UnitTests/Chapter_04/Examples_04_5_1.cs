#region Usings declarations

using Antlr4.Runtime;

using ApprovalTests;
using ApprovalTests.Reporters;

using Reefact.BookExamples.Antlr4.Chapter_04._5._1;

#endregion

namespace Reefact.BookExamples.Antlr4.UnitTests.Chapter_04 {

    [UseReporter(typeof(VisualStudioReporter))]
    public class Examples_04_5_1 {

        [Fact]
        public void display_column_of_index_0() {
            // Setup
            string           t_rows      = ResourcesHelper.Read("Chapter_04.t_xml");
            AntlrInputStream inputStream = AntlrInputStreamReader.Read(t_rows);
            GRun             grun        = GRun.Read(inputStream, 0);
            // Exercise
            string[] tokens = grun.GetTokens();
            // Verify
            Approvals.VerifyAll(tokens, "");
        }

    }

}