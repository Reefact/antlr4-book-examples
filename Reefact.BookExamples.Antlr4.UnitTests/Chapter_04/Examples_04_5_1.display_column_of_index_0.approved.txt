[0] = [@-1,0:0='<',<1>,1:0]
[1] = [@-1,1:5='tools',<10>,1:1]
[2] = [@-1,6:6='>',<5>,1:6]
[3] = [@-1,7:12='\r\n    ',<4>,1:7]
[4] = [@-1,13:13='<',<1>,2:4]
[5] = [@-1,14:17='tool',<10>,2:5]
[6] = [@-1,19:22='name',<10>,2:10]
[7] = [@-1,23:23='=',<7>,2:14]
[8] = [@-1,24:30='"ANTLR"',<8>,2:15]
[9] = [@-1,31:31='>',<5>,2:22]
[10] = [@-1,32:49='A parser generator',<4>,2:23]
[11] = [@-1,50:50='<',<1>,2:41]
[12] = [@-1,51:55='/tool',<9>,2:42]
[13] = [@-1,56:56='>',<5>,2:47]
[14] = [@-1,57:58='\r\n',<4>,2:48]
[15] = [@-1,59:59='<',<1>,3:0]
[16] = [@-1,60:65='/tools',<9>,3:1]
[17] = [@-1,66:66='>',<5>,3:7]
